view/adminhtml/templates/order/view/synced.phtml

```diff
@@ -15,24 +15,47 @@
  * @license    http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
  */
 
-// @codingStandardsIgnoreFile
-
-?>
-<?php
+/** @var \Taxjar\SalesTax\Block\Adminhtml\Order\View\Synced $block */
 $order = $block->getOrder();
-$orderSyncDate = '';
-
-if ($block->getSyncedAtDate($order)) {
+$synced = $block->getSyncedAtDate($order);
+if ($synced) {
     $orderSyncDate = $block->formatDate(
-        new \DateTime($block->getSyncedAtDate($order)),
+        new \DateTime($synced),
         \IntlDateFormatter::MEDIUM,
         true
     );
 }
 ?>
-<?php if (!empty($orderSyncDate)): ?>
+<?php if (isset($orderSyncDate)):?>
 <tr>
-    <th><?php /* @escapeNotVerified */ echo __('Synced to TaxJar') ?></th>
-    <td><?php /* @escapeNotVerified */ echo $orderSyncDate ?></td>
+    <th><?= /* @escapeNotVerified */ __('Synced to TaxJar') ?></th>
+    <td><?= /* @escapeNotVerified */ $orderSyncDate ?></td>
 </tr>
+<script>
+    require([
+        'jquery',
+    ], function($) {
+         window.syncTransaction = function(orderId) {
+             var syncButton = $('#taxjar_sync');
+             syncButton.prop('disabled', true);
+             syncButton.find('span').text('Syncing...');
+
+             $.ajax({
+                 url: '<?= $block->escapeUrl($block->getUrl('taxjar/transaction/sync')) ?>',
+                 method: 'POST',
+                 data: {
+                     order_id: orderId
+                 }
+             }).done(function(data) {
+                 if (data.result) {
+                     window.location.reload();
+                 } else {
+                     alert('Could not sync to TaxJar! Error: ' + data.error_message);
```

Comment:
I'm not sure that alerting here is the best way to handle errors. We already pass the exception message using the message manager ($this->messageManager->addErrorMessage).

Using alert (rather than just reloading the page regardless of result), can result in a scenario where the user can click sync multiple time. Each time it fails it adds a message to the message manager which can lead to this displaying when they reload the page (or even navigate to another page).

![image](https://user-images.githubusercontent.com/11321890/143097488-25b8b71a-aec9-4398-9270-54ec93c1160a.png)

Just reloading the page would also handle the scenarios where an the transaction is not synced but an exception is not thrown.

Reply:
This was me trying to be clever, but in retrospect, I'm really not sure why I didn't just reload page regardless of request outcome.. lol